### PR TITLE
Observer Mode

### PR DESCRIPTION
**Pull request by Quexington**
This PR is the first "catchup" into main for a feature from the long_lived/vault branch: observer mode.

With these changes, it will be possible to add just a pubkey to the keyring and wallets will sync balances as usual.  What this does _not_ add is the ability to sign transactions externally with the Chia Wallet, although the hooks to do so _are_ included so in theory external signers could begin to be developed (pending confirmation of CHIPs 27-31).  The Chia Wallet as a signer will likely be the next stage of "catchup"s from the long lived branch.

Changes from the following PRs are included in this diff:
* #16883 (superseded by #17715)
* #16895
* #16896
* #16967
* #16897
* #16903
* #17715

Additional changes after the initial commit [b79e2ea](https://github.com/Chia-Network/chia-blockchain/pull/17792/commits/b79e2eadb7fea9c596a76bc09d9ac8942c1e50cd) may be made due to review comments

### CODE REVIEW

**Review by altendky — DISMISSED** [review]

File: setup.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "zstd==1.5.5.1",
     "packaging==23.2",
     "psutil==5.9.4",
+    "hsms==0.3.1",
```
[the remark this answers is not in the document]
Comment (Quexington) [reply setup.py:39]: It is: https://github.com/Chia-Network/hsms

File: chia/cmds/keys.py [hunk 2]
```diff
@@ -73,12 +73,12 @@ def show_cmd(
     show_keys(ctx.obj["root_path"], show_mnemonic_seed, non_observer_derivation, json, fingerprint)
 
 
-@keys_cmd.command("add", help="Add a private key by mnemonic")
```
[the remark this answers is not in the document]
Comment (Quexington) [reply chia/cmds/keys.py:76]: This PR is not a "vault" PR despite coming from the long_lived branch of that name.  This change you've highlighted is actually kind of the crux and the rest of the changes are built around dealing with the implications of being able to add a public-key only entry.

File: build_scripts/check_dependency_artifacts.py [hunk 3]
```diff
@@ -9,6 +9,9 @@
 
 excepted_packages = {
     "dnslib",  # pure python
+    "chialisp_loader",
+    "chialisp_puzzles",
+    "chia_base",
```
[the remark this answers is not in the document]
Comment (altendky) [reply build_scripts/check_dependency_artifacts.py:14]: This had been discussed but I had forgotten.  This is tech debt and all packages should publish wheels even when they are pure Python.  Could we add a `TODO: publish wheels for these` so at least when someone looks at it it is clear that this shouldn't stay this way and how it should be addressed?

File: chia/rpc/util.py [hunk 4]
```diff
@@ -37,7 +43,13 @@ async def rpc_endpoint(self, request: Dict[str, Any], *args: object, **kwargs: o
             *args,
             **kwargs,
         )
-        return response_obj.to_json_dict()
+        if not request.get("chip-29", True):
```
[the remark this answers is not in the document]
Comment (Quexington) [reply chia/rpc/util.py]: Renamed

File: build_scripts/check_dependency_artifacts.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Quexington) [reply build_scripts/check_dependency_artifacts.py:15]: Added

**Review by wjblanke — DISMISSED** [review]
aok

**Comment by Quexington** [comment]
This is in draft because it probably should not go in while there is an outstanding release branch.

**Comment by wjblanke** [comment]
This is probably the reason for the coverage issue

https://github.com/Chia-Network/chia-blockchain/pull/17898

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**